src/components/TicketModal/TicketModal.vue

```diff
@@ -2,175 +2,268 @@
+    const selectedTag = ref({} as TagRecord);
+    const selectedTagId = ref([-1]);
+    const members = ref(Array<MemberRecord>());
+    const selectedUser = ref({} as UserRecord);
+    const selectedMember = ref({} as MemberRecord);
+    const statuses = ref(Array<StatusRecordOutput>());
+    const selectedStatus = ref({} as StatusRecordOutput);
+    const ticketObj = ref({} as WriteTicketRecord);
+    const doesTicketExist = ref(false);
     const shouldShowPencil = ref(true);
-    const ticketStatus = ref("In Progress");
-    const ticketUser = ref("Mason");
-    const ticketTeam = ref("Slugbotics");
-    const ticketTag = ref("Mechanical");
-    const ticketDueDate = ref("2018-07-22");
-    const statusColor = ref("#20A6EE");
-    const statusDropdownClass = ref("");
-    const testUsers = ref([{ data: "Mason" }, { data: "George" }]);
-    const testTeams = ref([{ data: "Slugbotics" }, { data: "Bugslotics" }]);
-    const testTags = ref([{ data: "Mechanical" }, { data: "Systems" }]);
-    const testStatuses = ref([
-      { data: "To Do" },
-      { data: "In Progress" },
-      { data: "Done" }
-    ]);
-    const statusSelected = (item: string) => {
-      ticketStatus.value = item;
+
+    const initializeData = () => {
+      if (props.ticketRecord) {
+        doesTicketExist.value = true;
+      }
+      selectedStatus.value.id = -1;
+      ticketObj.value.description = "";
+      ticketObj.value.title = "Title";
+      ticketObj.value.due_date = "";
     };
-    const userSelected = (item: string) => {
-      ticketUser.value = item;
+    const getTeams = async () => {
+      try {
+        teams.value = Object.values(await getUsersTeams());
+        selectedTeam.value.name = teams.value[0].name;
+        selectedTeam.value.id = teams.value[0].id;
+      } catch (error) {
+        alert(error);
+      }
     };
-    const teamSelected = (item: string) => {
-      ticketTeam.value = item;
+    const getTags = async () => {
+      try {
+        tags.value = Object.values(await listTags(selectedTeam.value.id));
+        selectedTag.value.title = "None";
+        selectedTagId.value = [-1];
+      } catch (error) {
+        alert(error);
+      }
     };
-    const tagSelected = (item: string) => {
-      ticketTag.value = item;
+    const getMembers = async () => {
+      try {
+        members.value = await listMembersDepaginated(selectedTeam.value.id);
+        selectedUser.value.username = "None";
+        selectedMember.value.id = "-1";
+      } catch (error) {
+        alert(error);
+      }
     };
-    const closeModal = () => {
-      context.emit("close");
+    const getStatuses = async () => {
+      try {
+        statuses.value = Object.values(
+          await listStatuses(selectedTeam.value.id)
+        );
+        selectedStatus.value.title = statuses.value[0].title;
+        selectedStatus.value.color = statuses.value[0].color;
+        selectedStatus.value.id = statuses.value[0].id;
```

Comment:
Is there a reason you need to assign each value individual instead of redefining the entire object. (ie `selectedStatus.value = statuses.value[0]`)

Reply:
fixed